src/ln/ln.rs

```diff
@@ -143,39 +143,45 @@ pub fn uumain(args: Vec<String>) -> i32 {
 }
 
 fn exec(files: &[PathBuf], settings: &Settings) -> i32 {
-    match settings.target_dir {
-        Some(ref name) => return link_files_in_dir(files, &PathBuf::from(name), &settings),
-        None => {}
+    if files.len() == 0 {
+        show_error!("missing file operand\nTry '{} --help' for more information.", NAME);
+        return 1;
+    }
+
+    // Handle cases where we create links in a directory first.
+    if let Some(ref name) = settings.target_dir {
+        // 4th form: a directory is specified by -t.
+        return link_files_in_dir(files, &PathBuf::from(name), &settings);
+    }
+    if !settings.no_target_dir {
+        if files.len() == 1 {
+            // 2nd form: the target directory is the current directory.
+            return link_files_in_dir(files, &PathBuf::from("."), &settings);
```

Comment:
Sorry it took so long for me to respond.  I think converting slashes to backslashes probably makes sense on Windows.